services/src/machine_learning/xgb_config_builder.rs

```diff
@@ -0,0 +1,1057 @@
+            _ => panic!("Unknown or unsupported booster type: {s}"),
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+pub enum Objective {
+    RegLinear,
+    #[default]
+    RegSquaredError,
+    RegLogistic,
+    BinaryLogistic,
+    BinaryLogisticRaw,
+    GpuRegLinear,
+    GpuRegLogistic,
+    GpuBinaryLogistic,
+    GpuBinaryLogisticRaw,
+    CountPoisson,
+    SurvivalCox,
+    MultiSoftmax,
+    MultiSoftprob,
+    RankPairwise,
+    RegGamma,
+    RegTweedie(Option<f32>),
+}
+
+impl ToString for Objective {
+    fn to_string(&self) -> String {
+        match *self {
+            Objective::RegLinear => "reg:linear".to_owned(),
+            Objective::RegSquaredError => "reg:squarederror".to_owned(),
+            Objective::RegLogistic => "reg:logistic".to_owned(),
+            Objective::BinaryLogistic => "binary:logistic".to_owned(),
+            Objective::BinaryLogisticRaw => "binary:logitraw".to_owned(),
+            Objective::GpuRegLinear => "gpu:reg:linear".to_owned(),
+            Objective::GpuRegLogistic => "gpu:reg:logistic".to_owned(),
+            Objective::GpuBinaryLogistic => "gpu:binary:logistic".to_owned(),
+            Objective::GpuBinaryLogisticRaw => "gpu:binary:logitraw".to_owned(),
+            Objective::CountPoisson => "count:poisson".to_owned(),
+            Objective::SurvivalCox => "survival:cox".to_owned(),
+            Objective::MultiSoftmax => "multi:softmax".to_owned(), // num_class conf must also be set
+            Objective::MultiSoftprob => "multi:softprob".to_owned(), // num_class conf must also be set
+            Objective::RankPairwise => "rank:pairwise".to_owned(),
+            Objective::RegGamma => "reg:gamma".to_owned(),
+            Objective::RegTweedie(_) => "reg:tweedie".to_owned(),
+        }
+    }
+}
+
+impl<'a> From<&'a str> for Objective {
+    fn from(s: &'a str) -> Self {
+        match s {
+            "reg:linear" => Objective::RegLinear,
+            "reg:squarederror" => Objective::RegSquaredError,
+            "reg:logistic" => Objective::RegLogistic,
+            "binary:logistic" => Objective::BinaryLogistic,
+            "binary:logitraw" => Objective::BinaryLogisticRaw,
+            "gpu:reg:linear" => Objective::GpuRegLinear,
+            "gpu:reg:logistic" => Objective::GpuRegLogistic,
+            "gpu:binary:logistic" => Objective::GpuBinaryLogistic,
+            "gpu:binary:logitraw" => Objective::GpuBinaryLogisticRaw,
+            "count:poisson" => Objective::CountPoisson,
+            "survival:cox" => Objective::SurvivalCox,
+            "multi:softmax" => Objective::MultiSoftmax,
+            "multi:softprob" => Objective::MultiSoftprob,
+            "rank:pairwise" => Objective::RankPairwise,
+            "reg:gamma" => Objective::RegGamma,
+            "reg:tweedie" => Objective::RegTweedie(None),
+            // TODO: better error handling
+            _ => panic!("no known objective function for {s}"),
```

Comment:
Implement TryFrom instead of From and return an error

Reply:
see: https://github.com/geo-engine/geoengine/pull/746#discussion_r1252302470